### PR TITLE
Add provider equinix and module_name in Equinix Metal TF configs

### PR DESCRIPTION
**Pull request by vasubabu**
What this PR does / why we need it:
It allows Vendor to declare the metadata fields. It adds provider meta module_name in Equinix Metal TF configs

Which issue(s) this PR fixes (optional, in fixes #(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):

Related to https://github.com/equinix/terraform-provider-equinix/issues/252

### CODE REVIEW

**Comment by ocobleseqx** [comment]
We need to merge https://github.com/equinix-labs/terraform-metal-hybrid-gateway/pull/6/ first and fix merge conflicts afterwards

**Comment by ocobleseqx** [comment]
#6 merged and conflicts resolved

**Comment by ctreatma** [comment]
@vasubabu there are merge conflicts on this PR.  Could you rebase your branch on the latest from `main`?

**Comment by ctreatma** [comment]
@vasubabu CI is failing due to a version constraint mismatch.  Please `git rm --cached .terraform.lock.hcl`, commit and push that up.  https://github.com/equinix-labs/terraform-metal-hybrid-gateway/actions/runs/4981698151/jobs/8916449175?pr=8

File: main.tf [hunk 1]
```diff
@@ -9,9 +9,15 @@ terraform {
...
-      version = "= 1.11.1"
+      version = "~> 1.14"
     }
   }
+  provider_meta "equinix" {
+    module_name = "equinix-hybrid-gateway"
+  }
+}
+provider "equinix" {
+  auth_token = var.auth_token
```
Comment (ctreatma) [inline main.tf:20]:
```suggestion
```

File: modules/backend/main.tf [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
   required_providers {
     equinix = {
-      source = "equinix/equinix"
+      source  = "equinix/equinix"
+      version = "~> 1.14"
     }
   }
+  required_version = ">= 0.13"
+  provider_meta "equinix" {
+    module_name = "equinix-fabric-connection"
```
Comment (ctreatma) [inline modules/backend/main.tf:10]:
```suggestion
    module_name = "equinix-hybrid-gateway/backend"
```

File: modules/ssh/main.tf [hunk 3]
```diff
@@ -1,9 +1,14 @@
...
   required_providers {
     equinix = {
-      source = "equinix/equinix"
+      source  = "equinix/equinix"
+      version = "~> 1.14"
     }
   }
+  required_version = ">= 0.13"
+  provider_meta "equinix" {
+    module_name = "equinix-hybrid-gateway"
```
Comment (ctreatma) [inline modules/ssh/main.tf:10]:
```suggestion
    module_name = "equinix-hybrid-gateway/ssh"
```

File: modules/frontend/main.tf [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
   required_providers {
     equinix = {
-      source = "equinix/equinix"
+      source  = "equinix/equinix"
+      version = "~> 1.14"
     }
   }
+  required_version = ">= 0.13"
+  provider_meta "equinix" {
+    module_name = "equinix-hybrid-gateway"
```
Comment (ctreatma) [inline modules/frontend/main.tf:10]:
```suggestion
    module_name = "equinix-hybrid-gateway/frontend"
```

**Review by ctreatma — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (vasubabu) [reply main.tf]: updated

**Review by ctreatma — APPROVED** [review]

**Pull request merged**